src/index.js

```diff
@@ -0,0 +1,898 @@
+     * Detect direction for an item to be selected next
+     * Key arrow down: index + 1
+     * Key arrow up: index - 1
+     */
+    const indexDelta = event.keyCode === this.KEYS.DOWN ? 1 : -1;
+
+    /**
+     * Getting search items
+     */
+    const searchItems = this.getSearchItems();
+    const selectedItem = this.getSelectedItem();
+
+    /**
+     * Get index of selected item
+     *
+     * @type {number}
+     */
+    const indexOfSelectedItem = searchItems.indexOf(selectedItem);
+
+    /**
+     * If there is selected item
+     */
+    if (indexOfSelectedItem !== -1) {
+      /**
+       * Magic math for getting correct index of item
+       *
+       * @type {number}
+       */
+      const nextIndex = (indexOfSelectedItem + indexDelta + searchItems.length) % searchItems.length;
+
+      /**
+       * Deselect prev item
+       */
+      searchItems[indexOfSelectedItem].classList.remove(this.CSS.searchItemSelected);
+
+      /**
+       * Highlight the next item
+       */
+      searchItems[nextIndex].classList.add(this.CSS.searchItemSelected);
+
+    /**
+     * If there is no search items was selected
+     */
+    } else {
+      /**
+       * If no search items then do nothing
+       */
+      if (!searchItems[0]) {
+        return;
+      }
+
+      /**
+       * Detect index for the next selected item depending the arrow key direction
+       *
+       * @type {number}
+       */
+      const nextIndex = indexDelta === 1 ? 0 : searchItems.length - 1;
+
+      /**
+       * Highlight the next item
+       */
+      searchItems[nextIndex].classList.add(this.CSS.searchItemSelected);
+    }
+  }
+
+  /**
+   * Process enter key pressing
+   *
+   * @returns
+   */
+  processEnterKeypressed() {
+    /**
+     * Try to get selected item
+     *
+     * @type {Element|null}
+     */
+    const selectedItem = this.getSelectedItem();
+
+    /**
+     * If any item was manually selected then process click on it
+     */
+    if (selectedItem) {
+      this.searchItemPressed(selectedItem);
+
+      return;
+    }
+
+    /**
+     * Get input field value
+     */
+    const href = this.nodes.inputField.value;
+
+    /**
+     * If input field is empty then do nothing
+     */
+    if (!href || !href.trim()) {
+      return;
+    }
+
+    /**
+     * If input is not a valid url then show an error
+     */
+    if (!url(href)) {
+      notifier.show({
+        message: DICTIONARY.invalidUrl,
+        style: 'error',
+      });
+
+      return;
+    }
+
+    /**
+     * Get the first item from the search list
+     * This item exists because input href is valid
+     *
+     * @type {Element}
+     */
+    const composedItem = this.getSearchItems()[0];
+
+    /**
+     * "Press" search item
+     */
+    this.searchItemPressed(composedItem);
+  }
+
+  /**
+   * Get search items
+   *
+   * @returns {Element[]}
+   */
+  getSearchItems() {
+    const nodesList = this.nodes.searchResults.querySelectorAll(`.${this.CSS.searchItem}`);
```

Comment:
Isn't `this.nodes.searchResults.children` the same?

Reply:
is think in this case, the searching by the classname is a more stable solution since the markup can be changed.